.devcontainer/Dockerfile

```diff
@@ -1,8 +1,20 @@
 FROM mcr.microsoft.com/dotnet/sdk:3.1-focal
+
+# Install Mono and Nuget CLI
+RUN apt-get -y update && \
+    apt-get -y install dirmngr gnupg apt-transport-https ca-certificates && \
+    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF && \
+    sh -c 'echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" > /etc/apt/sources.list.d/mono-official-stable.list' && \
+    apt-get -y update && \
+    apt-get -y install --no-install-recommends mono-complete nuget && \
+    apt-get clean && rm -rf /var/lib/apt/lists/
+
+# Install CMake, Ninja, LLVM/Clang tools
 RUN apt update \
     && apt-get install -y cmake \
     && apt-get install -y ninja-build \
     && apt-get install -y clang-11 \
     && apt-get install -y clang-tidy-11
```

Comment:
For reducing layer sizes, I'd recommend clearing the apt cache in the same layer as `apt update`. That does slow down building the devcontainer, but saves a lot of disk space.

Reply:
Good suggestion! I'll update this to incorporate clearing the cache.